Add links to relevant pages from config reference sections (#6586)
circleci/circleci-docs: jekyll/_cci2/configuration-reference.md

#### `environment`
{: #environment }
A map of environment variable names and values. For more information on defining and using environment variables, and the order of precedence governing the various ways they can be set, see the [Using Environment Variables]({{site.baseurl}}/docs/2.0/env-vars/) page.

#### `parallelism`
{: #parallelism }

> If `parallelism` is set to N > 1, then N independent executors will be set up and each will run the steps of that job in parallel. This features is used to optimize your test steps. Split your test suite, using the CircleCI CLI, across parallel containers so the job will complete in a shorter time. Certain parallelism-aware steps can opt out of the parallelism and only run on a single executor. Learn more on the [Running Tests in Parallel]({{ site.baseurl }}/2.0/parallelism-faster-jobs/) page.

Reviewer: ```suggestion
If `parallelism` is set to N > 1, then N independent executors will be set up and each will run the steps of that job in parallel. This feature is used to optimize your test steps. Split your test suite, using the CircleCI CLI, across parallel containers so the job will complete in a shorter time. Certain parallelism-aware steps can opt out of the parallelism and only run on a single executor. Learn more on the [Running Tests in Parallel]({{ site.baseurl }}/2.0/parallelism-faster-jobs/) page.
```

#### `parameters`
{: #parameters }

> `parameters` can be used when [calling that `job` in a `workflow`](#jobs-in-workflow).

Author: ```suggestion
`parameters` can be used when [calling a `job` in a `workflow`](#jobs-in-workflow).
```

entrypoint | N | String or List | The command used as executable when launching the container. `entrypoint` overrides the image's [`ENTRYPOINT`](https://docs.docker.com/engine/reference/builder/#entrypoint).
command | N | String or List | The command used as pid 1 (or args for entrypoint) when launching the container. `command` overrides the image's `COMMAND`. It will be used as arguments to the image `ENTRYPOINT` if it has one, or as the executable if the image has no `ENTRYPOINT`.
user | N | String | Which user to run commands as within the Docker container
environment | N | Map | A map of environment variable names and values
environment | N | Map | A map of environment variable names and values. The `environment` settings apply to the entrypoint/command run by the docker container, not the job steps.
auth | N | Map | Authentication for registries using standard `docker login` credentials
aws_auth | N | Map | Authentication for AWS Elastic Container Registry (ECR)
{: class="table table-striped"}

> The first `image` listed under a job defines the job's own primary container image where all steps will run.

Author: I've just moved this info into the table to make the whole thing a bit more succinct.

          aws_secret_access_key: $ECR_AWS_SECRET_ACCESS_KEY  # or project UI envar reference
```

> It is possible to reuse [declared commands]({{ site.baseurl }}/2.0/reusing-config/) in a job when using version 2.1. The following example invokes the `sayhello` command.

Author: This example was not relevant to this section.

**Note:** If you do not specify a resource class, CircleCI will use a default value that is subject to change.  It is best practice to specify a resource class as opposed to relying on a default.

**For self-hosted installations of CircleCI Server contact your system administrator for a list of available resource classes**.

> ##### Self-hosted runner

Author: ```suggestion
##### Self-hosted runner
{: #self-hosted-runner }
```

**Note:** If you do not specify a resource class, CircleCI will use a default value that is subject to change.  It is best practice to specify a resource class as opposed to relying on a default.

**For self-hosted installations of CircleCI Server contact your system administrator for a list of available resource classes**.

##### Self-hosted runner
{: #self-hosted-runner }

> Use the `resource_class` to configure a [self-hosted runner instance](https://circleci.com/docs/2.0/runner-overview/).
> 
> For example:
> 
> ```yaml
> jobs:
>   job_name:
>     machine: true
>     resource_class: my-namespace/my-runner
> ```

Author: ```suggestion
Use the `resource_class` key to configure a [self-hosted runner instance](https://circleci.com/docs/2.0/runner-overview/).

For example:

```yaml
jobs:
  job_name:
    machine: true
    resource_class: <my-namespace>/<my-runner>
```
```

###### **`filters`**
{: #jobfilters }

> Job Filters can have the key `branches` or `tags`. 

Author: ```suggestion
Job Filters can have the key `branches` or `tags`.
```
